changelog.d/13540.misc

```diff
@@ -0,0 +1 @@
+Add experimental configuration option to allow disabling legacy Prometheus metric names.
```

Comment:
At least for me an "experimental" configuration option means something very specific -- I don't think this is what we mean here?

```suggestion
Add a configuration option to allow disabling legacy Prometheus metric names.
```

Reply:
would you mind elaborating?

Reply:
"Experimental configuration" to me means "exists in the section `experimental_features` features portion of the configuration (and the corresponding `synapse.config.experimental` module).

Aka the stuff referred to in https://matrix-org.github.io/synapse/develop/development/experimental_features.html

Reply:
Ah, I mean 'experimental' in the literal sense here, in that it's an experiment and it's unproven, with possible failures. I don't necessarily want people moving to it right away.

Reply:
It wouldn't seem entirely unreasonable to put this setting under `experimental_features` anyway (except for the fact it feels more at home in the `metrics` section).

I think it's fine to call to call this an experimental option.